server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -637,30 +660,26 @@ public RestoreInProgress applyChanges(final RestoreInProgress oldRestore) {
         }
     }
 
-    public static RestoreInProgress.Entry restoreInProgress(ClusterState state, Snapshot snapshot) {
+    public static RestoreInProgress.Entry restoreInProgress(ClusterState state, String uuid) {
         final RestoreInProgress restoreInProgress = state.custom(RestoreInProgress.TYPE);
         if (restoreInProgress != null) {
-            for (RestoreInProgress.Entry e : restoreInProgress.entries()) {
-                if (e.snapshot().equals(snapshot)) {
-                    return e;
-                }
-            }
+            return restoreInProgress.entries().get(uuid);
         }
         return null;
     }
 
     static class CleanRestoreStateTaskExecutor implements ClusterStateTaskExecutor<CleanRestoreStateTaskExecutor.Task>, ClusterStateTaskListener {
 
         static class Task {
-            final Snapshot snapshot;
+            final String uuid;
```

Comment:
Identify `Task` by restore uuid now so we can have multiple snapshots from the same snapshot.